src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedCatchParameterShouldBeUnnamedCheck.java

```diff
@@ -0,0 +1,307 @@
+ *         reducing potential confusion and aiding in code maintenance.
+ *     </li>
+ * </ul>
+ * <p>
+ * See the <a href="https://docs.oracle.com/en/java/javase/21/docs/specs/unnamed-jls.html">
+ * Java Language Specification</a> for more information about unnamed variables.
+ * </p>
+ * <p>
+ * <b>Attention</b>: This check should be activated only on source code
+ * that is compiled by jdk21 or higher;
+ * unnamed catch parameters come out as the first preview in Java 21.
+ * </p>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code unused.catch.parameter}
+ * </li>
+ * </ul>
+ *
+ * @since 10.18.0
+ *
+ */
+
+@FileStatefulCheck
+public class UnusedCatchParameterShouldBeUnnamedCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_UNUSED_CATCH_PARAMETER = "unused.catch.parameter";
+
+    /**
+     * An array of unacceptable parent of ast of type {@link TokenTypes#IDENT}.
+     */
+    private static final int[] UNACCEPTABLE_PARENT_OF_IDENT = {
+        TokenTypes.DOT,
+        TokenTypes.LITERAL_NEW,
+        TokenTypes.METHOD_CALL,
+        TokenTypes.TYPE,
+    };
+
+    /**
+     * Keeps tracks of the current catch parameter.
+     */
+    private CatchParameter catchParameter;
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.LITERAL_CATCH,
+            TokenTypes.IDENT,
+        };
+    }
+
+    @Override
+    public void beginTree(DetailAST rootAST) {
+        catchParameter = null;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (ast.getType() == TokenTypes.LITERAL_CATCH) {
+            visitCatchParameter(ast);
+        }
+        else {
+            visitIdent(ast);
+        }
+    }
+
+    @Override
+    public void leaveToken(DetailAST ast) {
+        logUnusedCatchParameter(ast);
+    }
+
+    /**
+     * Visit ast of type {@link TokenTypes#LITERAL_CATCH}
+     * and set the catch parameter.
+     *
+     * @param catchAst token representing {@link TokenTypes#LITERAL_CATCH}
+     */
+    private void visitCatchParameter(DetailAST catchAst) {
+        final DetailAST catchParameterAst = catchAst.findFirstToken(TokenTypes.PARAMETER_DEF);
+        final String catchParameterName =
+                catchParameterAst.findFirstToken(TokenTypes.IDENT).getText();
+
+        if (!"_".equals(catchParameterName)) {
+            catchParameter = new CatchParameter(catchParameterName,
+                                    catchAst, catchParameterAst);
+        }
```

Comment:
Why not just assign this always, then check at the end if the catch parameter is unnamed?

Reply:
done